sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -178,13 +178,22 @@ public EventHubClientBuilder() {
      *     connection string.
      */
     public EventHubClientBuilder connectionString(String connectionString) {
-        final ConnectionStringProperties properties = new ConnectionStringProperties(connectionString);
-        final TokenCredential tokenCredential = new EventHubSharedKeyCredential(properties.getSharedAccessKeyName(),
-            properties.getSharedAccessKey(), ClientConstants.TOKEN_VALIDITY);
-
+        ConnectionStringProperties properties = new ConnectionStringProperties(connectionString);
+        TokenCredential tokenCredential = getTokenCredential(properties);
         return credential(properties.getEndpoint().getHost(), properties.getEntityPath(), tokenCredential);
     }
 
+    private TokenCredential getTokenCredential(ConnectionStringProperties properties) {
+        TokenCredential tokenCredential;
+        if (properties.getSharedAccessSignature() == null) {
+            tokenCredential = new EventHubSharedKeyCredential(properties.getSharedAccessKeyName(),
+                properties.getSharedAccessKey(), ClientConstants.TOKEN_VALIDITY);
+        } else {
+            tokenCredential = new EventHubSharedKeyCredential(properties.getSharedAccessSignature());
```

Comment:
Using constructor overloads work well. How about using a different sub class of `TokenCredential`, like `EventHubSharedAccessSignatureCredential`, or the shorter version `EventHubSASCredential`? It's cleaner to user two separate classes. Maybe someday the shared access signature will support new behaviors like token renew.

Reply:
This is similar to what .NET is doing. I am not sure we should introduce too many credential types when the token returned for both types of connection strings are the same.

Reply:
This is all internal. So I don't have very strong opinion to use separate classes. Let's go ahead with this.